src/ast-parser.y

```diff
@@ -592,35 +597,37 @@ block_instr :
     BLOCK labeling_opt block END labeling_opt {
       $$ = new_block_expr();
       $$->block = $3;
-      $$->block.label = $2;
-      CHECK_END_LABEL(@5, $$->block.label, $5);
+      $$->block->label = $2;
+      CHECK_END_LABEL(@5, $$->block->label, $5);
     }
   | LOOP labeling_opt block END labeling_opt {
       $$ = new_loop_expr();
       $$->loop = $3;
-      $$->loop.label = $2;
-      CHECK_END_LABEL(@5, $$->block.label, $5);
+      $$->loop->label = $2;
+      CHECK_END_LABEL(@5, $$->loop->label, $5);
     }
   | IF labeling_opt block END labeling_opt {
       $$ = new_if_expr();
       $$->if_.true_ = $3;
-      $$->if_.true_.label = $2;
-      CHECK_END_LABEL(@5, $$->block.label, $5);
+      $$->if_.true_->label = $2;
+      $$->if_.false_ = nullptr;
+      CHECK_END_LABEL(@5, $$->if_.true_->label, $5);
     }
   | IF labeling_opt block ELSE labeling_opt instr_list END labeling_opt {
       $$ = new_if_expr();
       $$->if_.true_ = $3;
-      $$->if_.true_.label = $2;
+      $$->if_.true_->label = $2;
       $$->if_.false_ = $6.first;
-      CHECK_END_LABEL(@5, $$->block.label, $5);
-      CHECK_END_LABEL(@8, $$->block.label, $8);
+      CHECK_END_LABEL(@5, $$->if_.true_->label, $5);
+      CHECK_END_LABEL(@8, $$->if_.true_->label, $8);
     }
 ;
 block :
     value_type_list instr_list {
-      WABT_ZERO_MEMORY($$);
-      $$.sig = $1;
-      $$.first = $2.first;
+      $$ = new Block();
+      $$->sig = std::move(*$1);
```

Comment:
Nit: Should the "delete $1" immediately follow the move, to make it clearer that a copy was constructed, and the old value should be deleted?

Note: Not that I asking for more changes, but at some point, you may want to change field "sig" to be a unique_ptr so that you don't have to build copies (just take over ownership), and let the destructor of what the unique_ptr appears in automatically delete the object.